src/rustup.rs

```diff
@@ -130,11 +130,26 @@ cross will not attempt to configure the toolchain further so that it can run you
     })
 }
 
-pub fn install_toolchain(toolchain: &QualifiedToolchain, msg_info: &mut MessageInfo) -> Result<()> {
+pub fn install_toolchain_command(
+    toolchain: &QualifiedToolchain,
+    msg_info: &mut MessageInfo,
+) -> Command {
     let toolchain = toolchain.to_string();
-    rustup_command(msg_info, false)
-        .args(&["toolchain", "add", &toolchain, "--profile", "minimal"])
+    let mut command = rustup_command(msg_info, false);
+    command.args(&["toolchain", "add", &toolchain, "--profile", "minimal"]);
+
+    command
+}
+
+pub fn install_toolchain(toolchain: &QualifiedToolchain, msg_info: &mut MessageInfo) -> Result<()> {
+    install_toolchain_command(toolchain, msg_info)
+        .arg("--force-non-host")
         .run(msg_info, false)
+        .or_else(|e| {
+            install_toolchain_command(toolchain, msg_info)
+                .run(msg_info, false)
+                .map_err(|_ignore| e)
```

Comment:
This now throws away this error instead.

Reply:
Should we wrap the error then? I felt it would produce redundant error messages.

Reply:
how about

```
.or_else(|e| {
            install_toolchain_command(toolchain, msg_info)
                .run(msg_info, false)
		.wrap_err("installing toolchain with and without `--force-non-host` failed")
```

Reply:
For better error handling, I've reverted to using version checking.